java/ql/src/experimental/Security/CWE/CWE-094/JexlInjectionLib.qll

```diff
@@ -0,0 +1,301 @@
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+
+/**
+ * A taint-tracking configuration for unsafe user input
+ * that is used to construct and evaluate a JEXL expression.
+ * It supports both JEXL 2 and 3.
+ */
+class JexlInjectionConfig extends TaintTracking::Configuration {
+  JexlInjectionConfig() { this = "JexlInjectionConfig" }
+
+  override predicate isSource(DataFlow::Node source) {
+    source instanceof TaintedSpringRequestBody or
+    source instanceof RemoteFlowSource
+  }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof JexlEvaluationSink }
+
+  override predicate isAdditionalTaintStep(DataFlow::Node fromNode, DataFlow::Node toNode) {
+    any(TaintPropagatingJexlMethodCall c).taintFlow(fromNode, toNode) or
+    returnsDataFromBean(fromNode, toNode)
+  }
+}
+
+/**
+ * A data flow source for parameters that have
+ * a Spring framework annotation indicating remote user input from servlets.
+ */
+private class TaintedSpringRequestBody extends DataFlow::Node {
+  TaintedSpringRequestBody() {
+    this.asParameter().getAnAnnotation() instanceof SpringServletInputAnnotation
+  }
+}
```

Comment:
This is not needed. It is already handled by `RemoteFlowSource`. The difference is that it actually isn't sufficient for a `@RequestMapping`-annotated method to have a parameter with a `@RequestParam` annotation - this can lead to false positives - the declaring class must also be annotated with `@Controller`. All of this complexity is modelled in `SpringController.qll`.

Reply:
(You can `git grep` for other tests that make use of `@RequestParam` sources to see examples of all the needed annotations.)